Update Go module dependencies, add StorageClient implementation, and introduce gRPC service for storage operations

jnathangreeg opened the pull request:
- Upgraded Go version to 1.25.0 and updated various dependencies in go.mod and go.sum.
- Implemented StorageClient with methods for sending container profiles and retrieving application/network profiles.
- Added gRPC service definitions and generated code for storage operations.
- Introduced test cases for StorageClient functionality.
- Updated constants for gRPC metadata keys in server configuration.

jnathangreeg committed:
Update Go module dependencies, add StorageClient implementation, and introduce gRPC service for storage operations

- Upgraded Go version to 1.25.0 and updated various dependencies in go.mod and go.sum.
- Implemented StorageClient with methods for sending container profiles and retrieving application/network profiles.
- Added gRPC service definitions and generated code for storage operations.
- Introduced test cases for StorageClient functionality.
- Updated constants for gRPC metadata keys in server configuration.

jnathangreeg committed:
Remove unnecessary instruction from Makefile regarding vendor directory after proto code generation

Naor-Armo committed:
Update pr.yaml

Signed-off-by: Naor-Armo <naorh@armosec.io>

jnathangreeg committed:
Add ListApplicationProfiles and ListNetworkNeighborhoods methods to StorageClient

- Implemented ListApplicationProfiles and ListNetworkNeighborhoods methods in StorageClient for retrieving application profiles and network neighborhoods.
- Added corresponding request and response message types in the proto definition.
- Updated mock client for testing and added unit tests for the new methods to ensure functionality and error handling.
- Enhanced gRPC service definitions to include new methods for application profiles and network neighborhoods.

jnathangreeg committed:
Merge branch 'feature/backend_storage' of github.com:kubescape/backend into feature/backend_storage

jnathangreeg committed:
Merge branch 'main' of github.com:kubescape/backend into feature/backend_storage

jnathangreeg committed:
Remove unused GrpcContextKey and related constant for customer GUID from consts.go

amirmalka commented on pkg/server/v1/consts.go:
```diff
@@ -35,4 +35,10 @@ const (
+	// GrpcAccessKeyHeader is the metadata key for access key authentication in gRPC calls
+	GrpcAccessKeyHeader = "x-api-token"
+	// GrpcAccountKey is the metadata key for customer GUID in gRPC calls
+	GrpcAccountKey = "customer-guid"
```
if we call it account.. why the value is customer-guid? 😅

amirmalka commented on pkg/client/v1/storageclient.go:
```diff
@@ -0,0 +1,413 @@
+	// Determine if connection should be secure
+	if c.grpcConfig != nil && c.grpcConfig.IsSecure {
+		// TODO: Add TLS credentials support for grpcs://
+		return fmt.Errorf("TLS support not yet implemented, use grpc:// scheme for insecure connections")
```
we don't support TLS?

jnathangreeg replied:
we do by the ingress

amirmalka commented on pkg/client/v1/storageclient.go:
```diff
@@ -0,0 +1,413 @@
+}
+
+// GetConnection returns the underlying gRPC connection (for advanced usage)
+func (c *StorageClient) GetConnection() *grpc.ClientConn {
```
not sure we need / want to expose conn

amirmalka commented on pkg/client/v1/storageclient.go:
```diff
@@ -0,0 +1,413 @@
+}
+
+// GetProtoClient returns the underlying proto client (for advanced usage)
+func (c *StorageClient) GetProtoClient() proto.StorageServiceClient {
```
same as https://github.com/kubescape/backend/pull/38/files#r2568344916

amirmalka commented on pkg/client/v1/storageclient.go:
```diff
@@ -0,0 +1,413 @@
+	}
+
+	req := &proto.GetProfileRequest{
+		Kind:         "networkNeighborhood",
```
```suggestion
		Kind:         "NetworkNeighborhood",
```

No?

amirmalka commented on pkg/client/v1/storageclient.go:
```diff
@@ -0,0 +1,413 @@
+	}
+
+	req := &proto.GetProfileRequest{
+		Kind:         "applicationProfile",
```
```suggestion
		Kind:         "ApplicationProfile",
```

jnathangreeg committed:
Refactor StorageClient to enhance gRPC metadata handling and update account key constant

- Introduced a new method `refreshMetadata` in StorageClient to rebuild gRPC metadata with current credentials.
- Updated the address formatting in NewStorageClient to use host and port.
- Changed the metadata key for account ID from "customer-guid" to "account-id" for clarity.
- Removed unused GetConnection and GetProtoClient methods to streamline the interface.

jnathangreeg committed:
fix test
